pkg/sync/syncWorker.go

```
@@ -193,15 +191,19 @@
+			conn, err := s.connectToNode(*node)
 			if err != nil {
 				continue
 			}
-			stream, err = s.setupStream(registration.ctx, *node, conn)
+
+			stream, err := s.setupStream(registration.ctx, *node, conn)
 			if err != nil {
+				_ = conn.Close()
 				continue
 			}
 			err = s.listenToStream(stream)
```

golangci-lint (1.62.2)

ineffectual assignment to err

(ineffassign)